changelog.d/20230911_113153_30907815+rjmello_HEAD.rst

```diff
@@ -0,0 +1,5 @@
+New Functionality
+^^^^^^^^^^^^^^^^^
+
+- The ``Executor`` will include the Python and SDK versions when registering
```

Comment:
Why only the 'Executor', and not just our SDK overall, even if using normal submit?

Reply:
I don't know if I fully understand the question, since `submit()` is an `Executor` method. If a user were to call `Client.register_function()` directly, they would have to enter the metadata as an argument.

Reply:
Sorry, my question was misleading/confusing.  I meant, normal registration now supports a 'metadata' field, and 'Executor' now uses that to add the two _version items in metadata.  The changelog meantions the latter, but not the former change that enables it.  Isn't it more complete to say something in the realm of "register_function() now supports metadata dict, and the executor uses it by default to add python and sdk version information"?

Reply:
Fair point. I've updated the changelog fragment in a fresh commit.